### PR TITLE
HornetQ Protocol Support

### PR DESCRIPTION
**Pull request by mtaylor**

### CODE REVIEW

**Comment by clebertsuconic** [comment]
Build failure...

The right name is Legacy Protocol Support...  I would avoid the H name now :8ball:

**Comment by clebertsuconic** [comment]
protocol/src/main/java/org/apache/activemq/artemis/core/protocol/hornetq/HQPropertiesConversionInterceptor.java:12: Using the '.*' form of import should be avoided - java.util.*.

File: artemis-protocols/artemis-hornetq-protocol/src/main/java/org/apache/activemq/artemis/core/protocol/hornetq/HQPropertiesConversionInterceptor.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
+package org.apache.activemq.artemis.core.protocol.hornetq;
```
Comment (johnament) [inline artemis-protocols/artemis-hornetq-protocol/src/main/java/org/apache/activemq/artemis/core/protocol/hornetq/HQPropertiesConversionInterceptor.java]: license header?

**Comment by clebertsuconic** [comment]
@mtaylor  this PR is not valid... there are a few checkstyle and a missing license header as @johnament  mentioned (although the PR check would have failed because of that).

You said you changed it.. but I don't see any changes.. the build still broken.. I just tried to merge it manually and these issues are still there.

**Comment by clebertsuconic** [comment]
@johnament  actually header check is disabled on the build.. we are fixing it back... thanks for raising the issue

File: artemis-protocols/artemis-hornetq-protocol/src/main/java/org/apache/activemq/artemis/core/protocol/hornetq/HQPropertiesConversionInterceptor.java (hunk 1 above)
Comment (mtaylor) [inline artemis-protocols/artemis-hornetq-protocol/src/main/java/org/apache/activemq/artemis/core/protocol/hornetq/HQPropertiesConversionInterceptor.java]: @johnament Thanks John.  This is now added.

**Comment by mtaylor** [comment]
@clebertsuconic Looks like IntelliJ settings are screwing things up.  I'll get to this tomorrow.

**Comment by clebertsuconic** [comment]
@mtaylor cool...  also please make sure you rebase on top of master tomorrow as I have merged a few other PRs today.

**Comment by mtaylor** [comment]
Please hold off merging this for now.  @andytaylor  has asked me to do some performance testing on the usage of interceptors vs overriding methods in the ProtocolManager.  I'll post back here with results.

**Comment by andytaylor** [comment]
personally I wouldn't have used Interceptors. Its a level of indirection that isn't really needed. I would just have used abstraction and catch the packets in extension points. It's easier to read and understand from a coding pov and we can be sure that performance isn't an issue or wont be an issue if someone changes the code. Saying that, if no one else has any misgivings then I'm happy for it to be merged.

**Comment by mtaylor** [comment]
The test consisted of sending a million empty messages 5x, taking average time.  The test is in no means 100% accurate as it was running on my dev laptop with other services running.  

The results show that running without interceptors, the test ran 700ms per million messages slower.  
This is obviously the lack of precision in the test as we would expect at least this to be equal or faster.

If we believe that this is a serious concern then we can run more accurate benchmarks.  However, I believe there is no significant over head in this approach.

Results are as follows:

=====================
Without Interceptors - Produce and Consume same session/thread:
Iteration Time: 58670
Iteration Time: 56232
Iteration Time: 55591
Iteration Time: 55013
Iteration Time: 55985
Average Time: 56298

Produce only:
Iteration Time: 19518
Iteration Time: 18240
Iteration Time: 13952
Iteration Time: 14758
Iteration Time: 17506
Average Time: 16794

======================

With Interceptors - Produce and Consume same session/thread:
Iteration Time: 59663
Iteration Time: 55300
Iteration Time: 54737
Iteration Time: 55259
Iteration Time: 53029
Average Time: 55597

Producer only:
Iteration Time: 19566
Iteration Time: 16898
Iteration Time: 13981
Iteration Time: 15014
Iteration Time: 14881
Average Time: 16068

**Comment by jbertram** [comment]
Martyn, can you rebase your PR?  There are some fixes merged recently that you need for the PR build to finish successfully.

**Comment by clebertsuconic** [comment]
There was a mistake on extra-tests/pom.xml... a comment that was removed by accident.

**Pull request closed without merging**